ush/forecast_postdet.sh

```diff
@@ -1,738 +1,414 @@
-  # Call child scripts in current script directory
+
   source "${USHgfs}/parsing_namelists_FV3.sh"
+  source "${USHgfs}/parsing_model_configure_FV3.sh"
+
   FV3_namelists
+  FV3_model_configure
+
   echo "SUB ${FUNCNAME[0]}: FV3 name lists and model configure file created"
 }
 
 FV3_out() {
   echo "SUB ${FUNCNAME[0]}: copying output data for FV3"
 
-  # Copy FV3 restart files
-  if [[ ${RUN} =~ "gdas" ]]; then
-    cd "${DATA}/RESTART"
-    local idate=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${restart_interval} hours" +%Y%m%d%H)
-    while [[ ${idate} -le ${forecast_end_cycle} ]]; do
-      for file in "${idate:0:8}.${idate:8:2}0000."*; do
-        ${NCP} "${file}" "${COM_ATMOS_RESTART}/${file}"
-      done
-      local idate=$(date --utc -d "${idate:0:8} ${idate:8:2} + ${restart_interval} hours" +%Y%m%d%H)
-    done
-  else
-    # No need to copy FV3 restart files when RUN=gfs or gefs
+  # Copy configuration files
+  if [[ "${RUN}" == "gfs" || "${RUN}" == "gefs" ]]; then
     ${NCP} "${DATA}/input.nml" "${COM_CONF}/ufs.input.nml"
     ${NCP} "${DATA}/model_configure" "${COM_CONF}/ufs.model_configure"
     ${NCP} "${DATA}/ufs.configure" "${COM_CONF}/ufs.ufs.configure"
     ${NCP} "${DATA}/diag_table" "${COM_CONF}/ufs.diag_table"
   fi
-  echo "SUB ${FUNCNAME[0]}: Output data for FV3 copied"
-}
 
-WW3_postdet() {
-  echo "SUB ${FUNCNAME[0]}: Linking input data for WW3"
-  COMPONENTwave=${COMPONENTwave:-${RUN}wave}
-
-  #Link mod_def files for wave grids
-  if [[ ${waveMULTIGRID} = ".true." ]]; then
-    local array=(${WAVECUR_FID} ${WAVEICE_FID} ${WAVEWND_FID} ${waveuoutpGRD} ${waveGRD} ${waveesmfGRD})
-    echo "Wave Grids: ${WAVECUR_FID} ${WAVEICE_FID} ${WAVEWND_FID} ${waveuoutpGRD} ${waveGRD} ${waveesmfGRD}"
-    local grdALL=$(printf "%s\n" "${array[@]}" | sort -u | tr '\n' ' ')
+  # Create an array of fv3 restart files
+  local fv3_restart_files tile_files fv3_restart_file restart_file
+  fv3_restart_files=(coupler.res fv_core.res.nc)
+  tile_files=(fv_core.res fv_srf_wnd.res fv_tracer.res phy_data sfc_data ca_data)
+  local nn tt
+  for (( nn = 1; nn <= ntiles; nn++ )); do
+    for tt in "${tile_files[@]}"; do
+      fv3_restart_files+=("${tt}.tile${nn}.nc")
+    done
+  done
 
-    for wavGRD in ${grdALL}; do
-      ${NCP} "${COM_WAVE_PREP}/${COMPONENTwave}.mod_def.${wavGRD}" "${DATA}/mod_def.${wavGRD}"
+  # Copy restarts in the assimilation window for RUN=gdas|enkfgdas|enkfgfs
+  if [[ "${RUN}" =~ "gdas" || "${RUN}" == "enkfgfs" ]]; then
+    local restart_date
+    restart_date=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+    while (( restart_date < forecast_end_cycle )); do
+      echo "Copying FV3 restarts for 'RUN=${RUN}' at ${restart_date}"
+      for fv3_restart_file in "${fv3_restart_files[@]}"; do
+        restart_file="${restart_date:0:8}.${restart_date:8:2}0000.${fv3_restart_file}"
+        ${NCP} "${DATArestart}/FV3_RESTART/${restart_file}" \
```

Comment:
We should use `cpfs` to copy these to COM (see #2349). Maybe later we'll changes `$NCP` to that, but for now:
```suggestion
        cpfs "${DATArestart}/FV3_RESTART/${restart_file}" \
```

Reply:
If you don't mind, I would like to defer this to a PR dedicated to this in itself.